### DISCUSSION TITLE
Limit full text search of specific content type

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#11163] [Q&A]
As i have conent type: Blog inside that there is alot of blog posts need to search inside of that content type of blog just, how can i limit full text search based on that content type?

As i see post full text search here:
https://docs.orchardcore.net/en/dev/docs/guides/implement-fulltext-search/

**Comment by Skrypt** [comment]
You create a new Lucene Index that you can name "BlogIndex" and on the index settings you select only the content type(s) you want to be indexed in that index. For example here you would only index BlogPost content type. The Full-Text settings will remain on each Content Type definition but the Index will only parse it for that specific content type. Then, you go to the "Configure Search Settings" admin page, select that new Lucene Index you created and you keep the "Content.ContentItem.FullText" as the "default searched fields".

Now, since you are only indexing "BlogPost" it should only return results of that content type from the index.

**Reply by saleemhaq** [reply]
> You create a new Lucene Index and on the index settings you select only the content type(s) you want to be indexed in that index. For example here you would only index BlogPost content type. The Full-Text settings will remain on each Content Type definition but the Index will only parse it for that specific content type. Then, you go to the "Configure Search Settings" admin page, select that new Lucene Index you created and you keep the "Content.ContentItem.FullText" as the "default searched fields".

I have to seperate the blog search and global search as it is for the whole application is that possible? so i create two Lucene Index?

**Reply by Skrypt** [reply]
Yes, that's what it is meant for.

**Reply by Skrypt** [reply]
But the Global search will now need to use your Global index, so you won't be able to use the current Search form to get global results if you set the "Configure Search Settings" to the new BlogIndex.
So, I suggest that you create a Blog search custom controller based on what we have in source code.

https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs

Change the code in that controller to always use your custom BlogIndex instead of getting the settings from the `siteSettings.As<LuceneSettings>();`.

You will need to have a custom route to something like /blog-search to get to the blog search page.

**Comment by Skrypt** [comment]
The issue here is that the Global Search Form can be used with only one Lucene Index at a time. And allowing to use a different Index from a QueryString param could be something to consider in the future, as long as we have permissions on these Indices it should be fine to allow overriding the default LuceneSettings from the Search Form itself.

**Reply by saleemhaq** [reply]
> The issue here is that the Global Search Form can be used with only one Lucene Index at a time. And allowing to use a different Index from a QueryString param could be something to consider in the future, as long as we have permissions on these Indices it should be fine to allow overriding the default LuceneSettings from the Search Form itself.

Is any other way to achieve that kind of scenario?

**Reply by Skrypt** [reply]
I've submitted a Pull Request here : https://github.com/OrchardCMS/OrchardCore/pull/11165
